modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -0,0 +1,1949 @@
+         */
+        _Tp at(size_t index) const;
+
+        /**
+         * @brief return the conjugate of this quaternion
+         * \f[q.conjugate() = (w, -x, -y, -z)\f]
+         */
+        Quat<_Tp> conjugate() const;
+
+        /**
+         * @brief return the value of exponential value
+         * \f[\exp(q) = e^w (\cos||\boldsymbol{v}||+ \frac{v}{||\boldsymbol{v}||})\sin||\boldsymbol{v}||\f]
+         * @param q a quaternion
+         *
+         * For example:
+         * ```
+         * Quatd q{1,2,3,4};
+         * cout << exp(q) << endl;
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> exp(const Quat<T> &q);
+
+        /**
+         * @brief return the value of exponential value
+         * \f[\exp(q) = e^w (\cos||\boldsymbol{v}||+ \frac{v}{||\boldsymbol{v}||}\sin||\boldsymbol{v}||)\f]
+         *
+         * For example
+         * ```
+         * Quatd q{1,2,3,4};
+         * cout << q.exp() << endl;
+         * ```
+         */
+        Quat<_Tp> exp() const;
+
+        /**
+         * @brief return the value of logarithm function
+         * \f[\ln(q) = \ln||q|| + \frac{\boldsymbol{v}}{||\boldsymbol{v}||}\arccos\frac{a}{||q||}\f]
+         * @param q a quaternion
+         * @param assumeUnit if true, q should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * Quatd q1{1,2,3,4};
+         * cout << log(q1) << endl;
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> log(const Quat<T> &q, bool assumeUnit);
+
+        /**
+         * @brief return the value of logarithm function
+         *  \f[\ln(q) = \ln||q|| + \frac{\boldsymbol{v}}{||\boldsymbol{v}||}\arccos\frac{w}{||q||}\f]
+         * @param assumeUnit if true, this quaternion should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * Quatd q(1,2,3,4);
+         * q.log();
+         *
+         * Quatd q1(1,2,3,4);
+         * q1.normalize().log(true);
+         * ```
+         */
+        Quat<_Tp> log(bool assumeUnit=false) const;
+
+        /**
+         * @brief return the value of power function with index \f$x\f$
+         * \f[q^x = ||q||(cos(x\theta) + \boldsymbol{u}sin(x\theta)))\f]
+         * @param q a quaternion
+         * @param x index of exponentiation
+         * @param assumeUnit if true, quaternion q should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * quatd q(1,2,3,4);
+         * power(q, 2);
+         *
+         * double angle = CV_PI;
+         * Vec3d axis{0, 0, 1};
+         * Quatd q1(angle, axis); //generate a unit quat by axis and angle
+         * power(q1, 2, true);
+         * ```
+         */
+        template <typename T, typename _T>
+        friend Quat<T> power(const Quat<T> &q, _T x, bool assumeUnit);
```

Comment:
It can be implemented without friendship, because there is public method  `power`.